### PR TITLE
API changes - table view wade data 

### PR DESCRIPTION
**Pull request by EverettRhodes**
- Backend changes to support TableView of data.
  - Does NOT include searching by Geometry. That will be a future PR.
- Added LinqKit to the Accessor project and used PredicateBuilder for creating the search query
  - used the extension pattern for predicate builder that was recommended by the LinqKit doc

### CODE REVIEW

File: src/API/WesternStatesWater.WestDaat.Tests.Helpers/AllocationAmountFactFaker.cs [hunk 1]
```diff
@@ -45,6 +67,20 @@ public static Faker<AllocationAmountsFact> SetAllocationExpirationDate(this Fake
             return faker;
         }
 
+        public static Faker<AllocationAmountsFact> SetExemptOfVolumeFlowPriority(this Faker<AllocationAmountsFact> faker, bool? isExempt)
+        {
+            if (isExempt != null)
+            {
+                faker
```
Comment (Rhipeus) [inline src/API/WesternStatesWater.WestDaat.Tests.Helpers/AllocationAmountFactFaker.cs:74]: Weird line break

**Review by Rhipeus — DISMISSED** [review]

File: src/API/WesternStatesWater.WestDaat.Accessors/Mapping/ApiProfile.cs [hunk 2]
```diff
@@ -53,6 +53,14 @@ public ApiProfile()
                 .ForMember(dest => dest.BeneficialUses, opt => opt.MapFrom(source => source.AllocationBridgeBeneficialUsesFact.Select(a => a.BeneficialUseCV)))
                 .ForMember(dest => dest.PriorityDate, opt => opt.MapFrom(source => source.AllocationPriorityDateNavigation.Date))
                 .ForMember(dest => dest.ExpirationDate, opt => opt.MapFrom(source => source.AllocationExpirationDateNavigation.Date));
+            CreateMap<EF.AllocationAmountsFact, WaterRightsSearchDetail>()
+                .ForMember(dest => dest.WadeUuid, opt => opt.MapFrom(source => source.AllocationAmountId.ToString()))
```
[the remark this answers is not in the document]
Comment (EverettRhodes) [reply src/API/WesternStatesWater.WestDaat.Accessors/Mapping/ApiProfile.cs:57]: yes. AllocationAmountID is a long and AllocationUuid is a string

File: src/API/WesternStatesWater.WestDaat.Accessors/EntityFramework/AllocationAmountsFact.cs [hunk 3]
```diff
@@ -69,5 +70,137 @@ public AllocationAmountsFact()
...
+
+            predicate = predicate.Or(x => x.AllocationBridgeSitesFact.Any(
+                sb => sb.Site.WaterSourceBridgeSitesFact.Any(
+                    wsb => waterSourceTypes.Contains(
+                        wsb.WaterSource.WaterSourceTypeCvNavigation.WaDEName.Length > 0 ? wsb.WaterSource.WaterSourceTypeCvNavigation.WaDEName : wsb.WaterSource.WaterSourceTypeCvNavigation.Name))));
+
+            return predicate;
+        }
+
+        public static ExpressionStarter<AllocationAmountsFact> HasOrginizationStates(List<string> states)
```
[the remark this answers is not in the document]
Comment (EverettRhodes) [reply src/API/WesternStatesWater.WestDaat.Accessors/EntityFramework/AllocationAmountsFact.cs:108]: fixed

File: src/API/WesternStatesWater.WestDaat.Accessors/EntityFramework/AllocationAmountsFact.cs [hunk 4]
```diff
@@ -69,5 +70,137 @@ public AllocationAmountsFact()
...
+            predicate = predicate.Or(x => states.Contains(x.Organization.State));
+
+            return predicate;
+        }
+
+        public static ExpressionStarter<AllocationAmountsFact> HasAllocationOwner(string allowcationOwner)
+        {
+            var predicate = PredicateBuilder.New<AllocationAmountsFact>();
+
+            predicate = predicate.Or(x => x.AllocationOwner == allowcationOwner);
```
[the remark this answers is not in the document]
Comment (EverettRhodes) [reply src/API/WesternStatesWater.WestDaat.Accessors/EntityFramework/AllocationAmountsFact.cs:121]: You are right.
fixed

File: src/API/WesternStatesWater.WestDaat.Accessors/EntityFramework/AllocationAmountsFact.cs [hunk 5]
```diff
@@ -69,5 +70,137 @@ public AllocationAmountsFact()
...
+
+            return predicate;
+        }
+
+        public static ExpressionStarter<AllocationAmountsFact> IsExemptOfVolumeFlowPriority(bool? isExempt)
```
[the remark this answers is not in the document]
Comment (EverettRhodes) [reply src/API/WesternStatesWater.WestDaat.Accessors/EntityFramework/AllocationAmountsFact.cs:126]: fixed

File: src/API/WesternStatesWater.WestDaat.Common/DataContracts/WaterRightsSearchDetail.cs [hunk 6]
```diff
@@ -0,0 +1,13 @@
+﻿namespace WesternStatesWater.WestDaat.Common.DataContracts
+{
+    public class WaterRightsSearchDetail
+    {
+        public string WadeUuid { get; set; }
+        public DateTime AllocationPriorityDate { get; set; }
```
[the remark this answers is not in the document]
Comment (EverettRhodes) [reply src/API/WesternStatesWater.WestDaat.Common/DataContracts/WaterRightsSearchDetail.cs:6]: fixed

File: src/API/WesternStatesWater.WestDaat.Common/DataContracts/WaterRightsSearchDetail.cs [hunk 7]
```diff
@@ -0,0 +1,13 @@
+﻿namespace WesternStatesWater.WestDaat.Common.DataContracts
+{
+    public class WaterRightsSearchDetail
+    {
+        public string WadeUuid { get; set; }
```
[the remark this answers is not in the document]
Comment (EverettRhodes) [reply src/API/WesternStatesWater.WestDaat.Common/DataContracts/WaterRightsSearchDetail.cs:5]: fixed

File: src/API/WesternStatesWater.WestDaat.Accessors/Mapping/ApiProfile.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (EverettRhodes) [reply src/API/WesternStatesWater.WestDaat.Accessors/Mapping/ApiProfile.cs:57]: fixed

File: src/API/WesternStatesWater.WestDaat.Accessors/WaterAllocationAccessor.cs [hunk 8]
```diff
@@ -1,21 +1,152 @@
...
         private readonly IDatabaseContextFactory _databaseContextFactory;
+        
+        public async Task<WaterRightsSearchResults> FindWaterRights(WaterRightsSearchCriteria searchCriteria)
+        {
+            var predicate = BuildWaterRightsSearchPredicate(searchCriteria);
+
+            using var db = _databaseContextFactory.Create();
+            var waterRightDetails = await db.AllocationAmountsFact
+                .Include(x => x.AllocationBridgeBeneficialUsesFact)
+                .Include(x => x.AllocationBridgeSitesFact)
```
[the remark this answers is not in the document]
Comment (EverettRhodes) [reply src/API/WesternStatesWater.WestDaat.Accessors/WaterAllocationAccessor.cs:33]: you are right. I probably added those before I had the projectTo in there
fixed

**Pull request merged**